foreman/data_refinery_foreman/surveyor/sra.py

```diff
@@ -264,25 +285,127 @@ def _build_file(run_accession: str, read_suffix="")-> File:
         metadata = SraSurveyor.gather_all_metadata(run_accession)
 
         if metadata["library_layout"] == "PAIRED":
-            files = [SraSurveyor._build_file(run_accession, "_1"),
-                     SraSurveyor._build_file(run_accession, "_2")]
+            files_urls = [SraSurveyor._build_file_url(run_accession, "_1"),
+                     SraSurveyor._build_file_url(run_accession, "_2")]
         else:
-            files = [SraSurveyor._build_file(run_accession)]
-
-        self.add_batch(platform_accession_code=metadata.pop("platform_instrument_model"),
-                       experiment_accession_code=metadata.pop("experiment_accession"),
-                       organism_id=int(metadata.pop("organism_id")),
-                       organism_name=metadata.pop("organism_name"),
-                       experiment_title=metadata.pop("experiment_title"),
-                       release_date=metadata.pop("run_ena_first_public"),
-                       last_uploaded_date=metadata.pop("run_ena_last_update"),
-                       files=files,
-                       key_values=metadata)
+            files_urls = [SraSurveyor._build_file_url(run_accession)]
+
+        ##
+        # Experiment
+        ##
+        experiment_accession_code = metadata.get('experiment_accession')
+        try:
+            experiment_object = Experiment.objects.get(accession_code=experiment_accession_code)
+            logger.error("Experiment %s already exists, skipping object creation.",
+                experiment_accession_code,
+                survey_job=self.survey_job.id)
+        except Experiment.DoesNotExist:
+            experiment_object = Experiment()
+            experiment_object.accession_code = experiment_accession_code
+            experiment_object.source_url = ENA_URL_TEMPLATE.format(experiment_accession_code)
+            experiment_object.source_database = "SRA"
+            experiment_object.platform_name = metadata.get("platform_instrument_model", "No model.")
+
+            # We don't get this value from the API, unfortunately.
+            # experiment_object.platform_accession_code = experiment["platform_accession_code"]
+
+            if not experiment_object.description:
+                experiment_object.description = "No description."
+
+            if "study_title" in metadata:
+                experiment_object.title = metadata["study_title"]
+            if "study_abstract" in metadata:
+                experiment_object.description = metadata["study_abstract"]
+            if "lab_name" in metadata:
+                experiment_object.submitter_institution = metadata["lab_name"]
+            if "experiment_design_description" in metadata:
+                experiment_object.protocol_description = metadata["experiment_design_description"]
+            if "pubmed_id" in metadata:
+                experiment_object.pubmed_id = metadata["pubmed_id"]
+                experiment_object.has_publication = True
```

Comment:
As far as we know, yes.

Reply:
Ok cool. I was just trying to figured out if `! has_publication == definitely_doesnt_have_publication` because that could be a potential nuance for the API to make clear.